csvcubed/csvcubed/schema/cube-config-schema.jsonc

```diff
@@ -1,303 +1,515 @@
-                    "type": "string"
+                "scaling_factor": {
+                  "description": "The scaling factor",
+                  "type": "decimal"
                 },
-
-				"definition_uri":  {
-                    "description": "The URI that defines the attribute source",
-                    "type": "string",
-                    "format": "uri"
+                "si_scaling_factor": {
+                  "description": "The SI scaling factor",
+                  "type": "decimal"
                 },
-				"required": {
-					"description": "optional, if set, the attribute requires values for each observation",
-					"type": "boolean"
-				},
-				"data_type": {
-					"description": "The data-type for the values for this attribute",
-					"type": "string"
-				},
-				"values": {
-					"description": "A list of the values for this attribute column",
-					"type": "array",
-					"items": {"$ref": "#/$defs/attributeValue"}
-				}
-			}
-        },
+                "quantity_kind": {
+                  "description": "Source URI for the definition of the Unit Kind",
+                  "type": "string",
+                  "format": "uri"
+                },
+                "definition_uri": {
+                  "description": "Source URI for the definition of the Unit",
+                  "$ref": "http://gss-cogs.github.io/family-schemas/external.json#/definitions/unit_ext_uri"
+                }
+              }
+            }
+          }
+          // units column Type
+        }
+      },
+      // columnTypes
 
-		"attributeValue": {
-			"description": "A single attribute value, optionally referencing existing values and definition URIs",
-			"type": "object",
-			"properties": {
-				"label": {
-					"type": "string",
-					"description": "A label for the attribute values"
-				},
-				"description": {
-					"type": "string",
-					"description": "The description of the attribute values"
-				},
-				"existing_value": {
-					"description": "The URI to an existing set of values",
-					"type": "string",
-					"format": "uri"
-				},
-				"definition_uri": {
-					"description": "The URI to an existing value definition",
-					"type": "string",
-					"format": "uri"
-				}
-			}
-		},
+      "valueTypes": {
+        "attribute": {
+          "description": "A single attribute value, optionally referencing existing values and definition URIs",
+          "type": "object",
+          "properties": {
+            "label": {
+              "type": "string",
+              "description": "A label for the attribute values"
+            },
+            "description": {
+              "type": "string",
+              "description": "The description of the attribute values"
+            },
+            "from_existing": {
+              "description": "The URI to an existing set of values",
+              "$ref": "http://gss-cogs.github.io/family-schemas/external.json#/definitions/attr_ext_uri"
```

Comment:
This isn't the right help to be providing to the user here. Attributes and attribute values are not the same kind of thing. Attribute values are something an attribute has whereas you're linking to the attributes themselves here. Leave it without intellisense/help for now.

Reply:
updated.